### ISSUE TITLE
Android: No client found with id X

### ISSUE THREAD

**Issue by seth-shi** [tradle/react-native-udp#118]
```
socket-0 binding, address: 0.0.0.0 port: 5607
 LOG  socket-0 failed to bind {"message": "no client found with id 0"}
```

Only launch the app for the first time every time, After the normal

**Comment by gibo77** [comment]
Share your code so we can help you. 


Sent from Yahoo Mail for iPhone


On Sunday, May 31, 2020, 3:55 AM, 施国鹏 <notifications@github.com> wrote:



socket-0 binding, address: 0.0.0.0 port: 5607
 LOG  socket-0 failed to bind {"message": "no client found with id 0"}

Only launch the app for the first time every time, After the normal

**Comment by seth-shi** [comment]
this my code.
```
async componentDidMount() {

        // origin 
        // let udpPort = await Storage.udpPort();
        // udpPort = parseInt(udpPort);
        // test random port
        let udpPort = parseInt(Math.random() * 8000 + 5000);

        try {

            const udp = UdpSocket.createSocket('udp4');
            udp.bind({port: udpPort});

            this.setState({logs: LogUtil.info(i18n.t('udpServerStart'), udpPort) + this.state.logs});

            udp.on('message', (msgArr, rinfo) => {

                if (typeof rinfo !== 'object' || !rinfo.address) {
                    return;
                }

                const host = rinfo.address;
                if (this.state.hosts[host] === undefined) {

                    const msg = msgArr.map(char => {
                        return String.fromCodePoint(char);
                    }).join('');


                    const hostObj = new Host(host, msg, false);
                    const hosts = Object.assign({}, this.state.hosts, {host: hostObj});

                    this.setState({
                        hosts: hosts,
                        logs: LogUtil.info(i18n.t('receiveBroadcast'), 'host:', hostObj.ip, ' port:' + hostObj.port) + this.state.logs,
                    });
                }
            });

            this._udp = udp;

        } catch (e) {


            this.setState({
                logs: LogUtil.info(i18n.t('udpServerStartFail'), '\n', e.message) + this.state.logs,
            });
        }


    }
```

**Comment by Fr33maan** [comment]
I can confirm without being able to reproduce it in a consitent way. @Rapsssito could you tell me if it's something you experienced ?
I've seen it in android, currently writing end 2 end tests for production build which made me able to spot it.

**Comment by Rapsssito** [comment]
@Fr33maan, I can confirm. I have merged a PR (#108) with a possible fix, but it has not been published yet.

I am still waiting @mvayngrib for setting up a NPM_TOKEN secret token for semantic release.

**Comment by Fr33maan** [comment]
Nice, right now I'm using an ugly fix. Glad it will land on npm soon ! @mvayngrib sorry for the spam but it's necessary <3

**Comment by alex-ferguson-spi** [comment]
I am still having this issue where the error is saying "no client found with id 0" on the first time of opening a socket. This is what my code looks like
```
import dgram from "react-native-udp";
let socket = dgram.createSocket({ type: "udp4", reuseAddr: true });
socket.bind(6789);
```

I am using this with React Native v.63 and react-native-udp ^3.1.0 on a physical Samsung android tablet.

**Comment by alex-ferguson-spi** [comment]
> @Fr33maan, I can confirm. I have merged a PR (#108) with a possible fix, but it has not been published yet.
> 
> I am still waiting @mvayngrib for setting up a NPM_TOKEN secret token for semantic release.

I can confirm that this worked. I monkey wrenched the changes into the node module and so far after 3 tests where I fully restarted everything (metro, the tablet, reinstalling the app) I no longer have the no client found with id 0 issue anymore

**Comment by Rapsssito** [comment]
This issue should be fixed by [v3.2.0](https://github.com/tradle/react-native-udp/tree/v3.2.0)!

**Comment by hubciorz** [comment]
@Rapsssito Unfortunately, it's not :( It happens from time to time and directly just before these ones happen: #119 and #143
